R/dialogs.R

```diff
@@ -0,0 +1,110 @@
+#' Show Dialog Box
+#'
+#' Shows a dialog box with a given title and contents.
+#'
+#' @param title The title to display in the dialog box.
+#'
+#' @param message A character vector witht he contents to display in
+#'   the main dialog area. Contents can contain the following HTML tags:
+#'   "p", "em", "strong", "b" and "i".
+#'
+#' @param url And optional url to display under the \code{message}.
+#'
+#' @details
+#'
+#' \preformatted{
+#'     showDialog("A dialog", "Showing <b>bold</b> text in the message.")
+#' }
+#'
+#' @export
+showDialog <- function(title, message, url = NULL) {
+  callFun("showDialog", title, message, url)
+}
+
+#' Updates a Dialog Box
+#'
+#' Updates specific properties from the current dialog box.
+#'
+#' @param ... Named parameters and values to update a dialog box.
+#'
+#' @details
+#'
+#' Currently, the only dialog with support for this action is the
+#' New Connection dialog in which the code preview can be
+#' updated through this API.
+#'
+#' \preformatted{
+#'     updateDialog(code = "con <- NULL")
+#' }
+#'
+#' @export
+updateDialog <- function(...) {
+  callFun("updateDialog", ...)
+}
+
+#' Show Prompt Dialog Box
+#'
+#' Shows a dialog box with a prompt field.
+#'
+#' @param title The title to display in the dialog box.
+#'
+#' @param message A character vector witht he contents to display in
+#'   the main dialog area.
+#'
+#' @param default An optional character vector that fills the prompt field
+#'   with a default value.
+#'
+#' @export
+showPrompt <- function(title, message, default = NULL) {
+  callFun("showPrompt", title, message, default)
+}
+
+#' Show Question Dialog Box
+#'
+#' Shows a dialog box asking a question.
+#'
+#' @param title The title to display in the dialog box.
+#'
+#' @param message A character vector witht he contents to display in
+#'   the main dialog area.
+#'
+#' @param ok And optional character vector that overrides the caption for
+#'   the OK button.
+#'
+#' @param cancel An optional character vector that overrides the caption for
+#'   the Cancel button.
+#'
+#' @details
+#'
+#' @export
+showQuestion <- function(title, message, ok = NULL, cancel = NULL) {
+  callFun("showQuestion", title, message, ok, cancel)
+}
+
+#' Read Preference
```

Comment:
Okay, I get why these are exposed now (dialog UI remembering prefs). I think it's okay to expose these but we should "namespace" them sort of like we did here for persistent values: https://github.com/rstudio/rstudio/commit/47ae3dc69090ccd3e31c42a4d63b0b5c3fc8e9f8